File: Remove loop in Linked List.java
class Solution {
    // Function to remove a loop in the linked list.
    public static void removeLoop(Node head) {
        // code here
        Node cur=head;
        Set<Node> res = new HashSet<Node>();
        while(cur!=null){
            if(res.contains(cur.next)){
                cur.next=null;
            }
            res.add(cur);
            cur = cur.next;
        }
    }
}

File: Remove loop in Linked List.py
class Solution:
    #Function to remove a loop in the linked list.
    def removeLoop(self, head):
        # code here
        res=set()
        cur=head
        while cur is not None:
            if cur.next in res:
                cur.next = None
                return
            res.add(cur)
            cur=cur.next
